pkg/conduit/pipelines_parser_test.go

```diff
@@ -145,7 +145,62 @@ func Test_Parser_Invalid_Version(t *testing.T) {
 		t.Error(err)
 	}
 
-	p, err := Parse(bytes.NewReader(data))
+	p, err := Parse(data)
 	assert.Error(t, err)
 	assert.Equal(t, p, PipelinesInfo{})
 }
+
+func Test_Parser_Env_Vars(t *testing.T) {
+	filename, err := filepath.Abs("./test/pipelines4.yml")
+	if err != nil {
+		t.Error(err)
+	}
+
+	// set env variables
+	err = os.Setenv("AWS_SECRET", "my-aws-secret")
+	if err != nil {
+		t.Fatalf("Failed to write env var: $AWS_SECRET")
+	}
+	err = os.Setenv("AWS_KEY", "my-aws-key")
+	if err != nil {
+		t.Fatalf("Failed to write env var: $AWS_KEY")
+	}
+	err = os.Setenv("AWS_URL", "aws-url")
+	if err != nil {
+		t.Fatalf("Failed to write env var: $AWS_URL")
+	}
```

Comment:
It might be good to clean up these. If they are used when normally developing something (e.g. when working on the S3 connector), they may interfere. Another option is to use some bogus or random variable names.

Reply:
These env vars will only be set for the lifetime of the test run (i.e. one invocation of `go test`). Granted, they will be visible in any test that is run after this test, but they shouldn't be visible after `go test` stops running, so not a lot of harm is done, I don't think other tests use these variables. That said, I agree that we could maybe use var names that we know won't interfere with anything like `CONDUIT_TEST_PARSER_X` or something.